src/sortedDict.jl

```diff
@@ -256,22 +260,55 @@ immutable SDKeyIteration{T <: SDIterableTypesBase}
     base::T
 end
 
+eltype{K,D,Ord <: Ordering}(s::SDKeyIteration{SortedDict{K,D,Ord}}) = K
+eltype{K,D,Ord <: Ordering}(s::SDKeyIteration{ExcludeLast{K,D,Ord}}) = K
+eltype{K,D,Ord <: Ordering}(s::SDKeyIteration{IncludeLast{K,D,Ord}}) = K
```

Comment:
My point was that `Array{Complex}` is a subtype of with `Array` but `Array{Complex{Float64}}` is not a subtype of `Array{Complex}`.

```julia
julia> type A{T1, T2}
       end

julia> type B{T}
       end

julia> f{T}(::B{A{T}}) = T
f (generic function with 1 method)

julia> f(B{A{Int, Int}}())
ERROR: MethodError: `f` has no method matching f(::B{A{Int64,Int64}})

julia> f(B{A{Int}}())
Int64
```